tests/services/test_cluster.py

```diff
@@ -73,40 +73,51 @@ def _get_cluster() -> Cluster:
                     "10.0.0.2",
                 ],
             },
-            [call({}, "10.0.0.1"), call({}, "10.0.0.2"), call({}, None)],
+            [({}, "10.0.0.1"), ({}, "10.0.0.2"), ({}, None)],
             None,
         ),
     ],
 )
 @pytest.mark.asyncio
 @pytest.mark.skipif(sys.version_info < (3, 8), reason="AsyncMock requires python 3.8+")
-async def test_spawn_instances(kwargs, calls, error):
+async def test_spawn_instances(kwargs, args, error, monkeypatch):
+    def _get_new_engine(self):
+        return mock.AsyncMock()
+
+    monkeypatch.setattr(Golem, "_get_new_engine", _get_new_engine)
+
     with patch("yapapi.services.Cluster.spawn_instance") as spawn_instance:
-        cluster = _get_cluster()
+        golem = Golem(budget=1)
```

Comment:
hmm... not nitpicking just asking out of curiosity -> wasn't it possible to leave the smaller footprint of the unittest which does not include `Golem`? ...

Reply:
1. It would be pretty hard since we don't have `Cluster.spawn_instances` anymore
2. I think this sort of a test that is somewhere between unit and integration is useful here. Yes, it's more complex, but we're still testing what is most important for us.